app/backend/src/clj/teet/file/file_commands.clj

```diff
@@ -85,6 +86,28 @@
     {:url (file-storage/upload-url key)
      :file (d/pull (:db-after res) '[*] file-id)}))
 
+(defcommand :file/upload-user-attachment
+  {:doc "Upload attachment file and optionally attach it to entity."
+   :context {:keys [conn user db]}
+   :payload {:keys [file employee-id]}
+   :project-id nil
+   :authorization {:contracts/contract-editing {}}
+   :pre {}}
+  (log/debug "upload-user-attachment: employee " employee-id)
+
+  (let [key (new-file-key file)
+        res (tx [(merge (select-keys file file-keys)
+                          {:db/id "new-file"
+                           :file/s3-key key}
+                          (creation-meta user))])
+        file-id (get-in res [:tempids "new-file"])
+        _ (tx [{:db/id (contract-db/get-user-for-company-contract-employee db employee-id)
+                :user/files file-id}
+               {:db/id employee-id
+                :company-contract-employee/attached-files file-id}])]
```

Comment:
I'm not 100% sure, but I think you can combine the two transactions above by using the `"new-file"` placeholder also for `:user/files`.

Reply:
Works! Thank you!